tfs/src/main/java/hudson/plugins/tfs/JenkinsEventNotifier.java

```diff
@@ -62,6 +68,46 @@ public static void sendJobCompletionEvent(final JSONObject payload) {
     }
 
     /**
+     * Send the build status of VSTS pull request back to connected TFS/VSTS servers only for those jobs kicked off by PRs.
+     */
+    public static void sendPullRequestBuildStatusEvent(final CommitParameterAction commitParameter, final GitStatusState buildState, final String buildDescription, final String targetUrl, final String jobFullPath) {
+        try {
+            if (commitParameter != null) {
+                if (commitParameter instanceof PullRequestParameterAction) {
+                    final PullRequestParameterAction prpa = (PullRequestParameterAction) commitParameter;
+                    final PullRequestMergeCommitCreatedEventArgs pullRequestMergeCommitCreatedEventArgs = prpa.getPullRequestMergeCommitCreatedEventArgs();
+                    sendPullRequestBuildStatusEvent(pullRequestMergeCommitCreatedEventArgs, buildState, buildDescription, targetUrl, jobFullPath);
+                }
+            }
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+        return;
+    }
+
+    /**
+     * Send the build status of VSTS pull request back to connected TFS/VSTS servers.
+     */
+    public static void sendPullRequestBuildStatusEvent(final PullRequestMergeCommitCreatedEventArgs gitCodePushedEventArgs, final GitStatusState buildState, final String buildDescription, final String targetUrl, final String jobFullPath) {
+        try {
+            final TeamGitStatus status = new TeamGitStatus();
+            status.state = buildState;
+            status.description = buildDescription;
+            status.targetUrl = targetUrl;
+            status.context = new GitStatusContext(" ", StringUtils.stripEnd(jobFullPath, "/"));
+
+            final URI collectionUri = gitCodePushedEventArgs.collectionUri;
+            final TeamRestClient client = new TeamRestClient(collectionUri);
+            client.addPullRequestStatus(gitCodePushedEventArgs, status);
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
These errors here should be sent to the logger (https://github.com/jenkinsci/tfs-plugin/pull/175/files#diff-462c46539e78a70b95caa4cf430ecbd7R39) so they are easy to find and filter

Reply:
Fixed.